ggml/src/ggml-backend.cpp

```diff
@@ -770,12 +772,21 @@ static const char * ggml_backend_cpu_buffer_type_get_name(ggml_backend_buffer_ty
 }
 
 static ggml_backend_buffer_t ggml_backend_cpu_buffer_type_alloc_buffer(ggml_backend_buffer_type_t buft, size_t size) {
+#if defined(GGML_USE_METAL) || defined(TARGET_OS_OSX)
```

Comment:
We cannot rely on backend macros like `GGML_USE_METAL`, and I don't see the point of enabling this specifically with Metal anyway, when the crash is happening in the CPU backend. Instead I would suggest using `GGML_ALIGNED_MALLOC` everywhere like the TODO in the comment suggests.

Reply:
The main focus here is on `TARGET_OS_OSX`, but I figured that including `GGML_USE_METAL` covers more cases where `posix_memalign` is available and can be used instead of `malloc`.

I'll move `GGML_ALIGNED_MALLOC` to `ggml-backend-impl.h` and use it instead then.

Reply:
It still needs some changes:
- It should be in `ggml-impl.h`, not `ggml-backend-impl.h`
- The implementation details need to be hidden in the source file, the header should only have the `ggml_aligned_malloc` and `ggml_aligned_free` functions, and the macros should be removed

I realize this is a lot more work than just fixing the bug, but we cannot add technical debt every time we fix a bug.